src/test/language-server-test.ts

```diff
@@ -346,4 +346,176 @@ describe('LSP', function () {
+                    'c.ts': '/// <reference path="d.ts"/>\nexport function foo() {}',
+                    'd.ts': 'export function bar() {}',
+                },
+                'missing': {
+                    'a.ts': '/// <reference path="b.ts"/>\n/// <reference path="missing.ts"/>\nnamespace t {\n    function foo() : Bar {\n        return null;\n    }\n}',
+                    'b.ts': 'namespace t {\n    export interface Bar {\n        id?: number;\n    }}'
+                }
+            }, done);
+        });
+        it('definition', function (done: (err?: Error) => void) {
+            utils.definition({
+                textDocument: {
+                    uri: 'file:///a.ts'
+                },
+                position: {
+                    line: 2,
+                    character: 0
+                }
+            }, {
+                    uri: 'file:///foo/c.ts',
+                    range: {
+                        start: {
+                            line: 0,
+                            character: 0
+                        },
+                        end: {
+                            line: 0,
+                            character: 24
+                        }
+                    }
+                }, () => {
+                    utils.definition({
+                        textDocument: {
+                            uri: 'file:///a.ts'
+                        },
+                        position: {
+                            line: 3,
+                            character: 2
+                        }
+                    }, {
+                            uri: 'file:///foo/d.ts',
+                            range: {
+                                start: {
+                                    line: 0,
+                                    character: 0
+                                },
+                                end: {
+                                    line: 0,
+                                    character: 24
+                                }
+                            }
+                        }, done);
+                });
+        });
+        it('definition with missing ref', function (done: (err?: Error) => void) {
+            utils.definition({
+                textDocument: {
+                    uri: 'file:///missing/a.ts'
+                },
+                position: {
+                    line: 3,
+                    character: 21
+                }
+            }, {
+                    uri: 'file:///missing/b.ts',
+                    range: {
+                        start: {
+                            line: 1,
+                            character: 4
+                        },
+                        end: {
+                            line: 3,
+                            character: 5
+                        }
+                    }
+                }, done);
+        });
+        it('TODO deep definition', function (done: (err?: Error) => void) {
```

Comment:
This test passes but should fail because we do not support definition resolution for depths > 3 (A refers to B, B refers to C, C refers to D, D defines something). There are samples in microsoft/typescript code. I'd like to remove TODO when we'll deal what to do with depth limitation